app/src/main/java/com/amaze/filemanager/filesystem/compressed/CompressedHelper.java

```diff
@@ -118,13 +118,14 @@ public static Extractor getExtractorInstance(
     } else if (isBzip2(type)) {
       extractor = new Bzip2Extractor(context, file.getPath(), outputPath, listener, updatePosition);
     } else {
-      return null;
+      throw new IllegalArgumentException("The current compressed file has no way of opening it!" + file);
```

Comment:
The crash report points to getCompressorInstance method but the change is made in getExtractInstance.
Also to point out, throwing an illegal argument exception isn't user friendly, either throw a toast message saying 'operation failed'

Reply:
> Also to point out, throwing an illegal argument exception isn't user friendly, either throw a toast message saying 'operation failed'

The problem is checking if the decompressor can decompress isn't this function's resposability, that resposability belies in `isFileExtractable`. If this crashes, it means someone forgot to add a case here or forgot to check if the compressed file is actually extractable.

Reply:
Also, we have no way to track down the issue if the error is "operation failed".

Reply:
Sorry I should rephrase. thowing illegal argument exception only isn't enough here. This'll still be a crash page for user.
I do understand your point, 'operation failed' doesn't give better understanding to us, that's why we log this crash wherever this method call is used and show a toast to user instead. 
In future I do have plans to extract all logs in a session to a log file and allow user to send us that logfile. That'll allow us to find the exact cause for this and other similar 'operation failed' scenarios around the app.

Reply:
Hmm, I'll make it a crash if DEBUG and a toast if not, but I'll have to see how to make it work.